Feature/filtering flights

LamberM opened the pull request.

kbannach commented on src/main/java/com/findmyflight/findmyflight/service/flightwatcher/CreateOrUpdateFlightWatcherRequest.java:
```diff
@@ -13,6 +15,7 @@ public record CreateOrUpdateFlightWatcherRequest(
         LocalDate fromDate,
         LocalDate toDate,
         @NotNull Boolean suspended,
+        @NotNull @Min(value = 1, message = "maxPrice should be more than 0") BigDecimal maxPrice,
```
this parameter is optional, so `@NotNull` is incorrect

kbannach replied:
also, this field is not being used yet in `BiletyLotniczeFlightsReader`, and that's also the part of this story - we need to filter search results with this `maxPrice` parameter. So there's a bit of implementation still to be done

kbannach replied:
this is also true to a newly added `maxFlightCount` field

kbannach commented on src/main/java/com/findmyflight/findmyflight/service/emailnotificationreceiver/EmailNotificationReceiver.java:
```diff
@@ -30,5 +31,6 @@ public class EmailNotificationReceiver {
     private String address;
 
     @OneToMany(mappedBy = "emailNotificationReceiver")
+    @Size(max = 20)
```
why do we need this?

kbannach commented on src/main/java/com/findmyflight/findmyflight/service/flightwatcher/FlightWatcher.java:
```diff
@@ -46,6 +47,9 @@ public class FlightWatcher {
     @NotNull
     private Boolean suspended;
 
+    @NotNull
```
as on the controller, this can be `null`
also, validation should be done on controller AND ALSO on database, so `@Min` annotation should also be used here. This will translate to some column constraints in the DB schema, example log when running tests:
```
Hibernate: 
    create table flight_watcher (
        from_date date,
        max_price numeric(38,2) not null check (max_price>=0),
        suspended boolean not null,
        to_date date,
        email_notification_receiver_id bigint not null,
        id bigserial not null,
        from_city varchar(255),
        to_city varchar(255),
        primary key (id)
    )
```

kbannach commented on src/main/java/com/findmyflight/findmyflight/service/flightwatcher/FlightWatcherService.java:
```diff
@@ -59,4 +69,8 @@ public void suspend(Long id) {
         flightWatcherRepository.resume(id);
     }
+
+    private boolean maxFlightCountIsValid(Long id) {
```
this is not what we mean when talking about "max flights count". This feature is not about throwing errors when we find too many flights, but it's about enabling our users to limit the flight result count. I think I could've described it more in detail when creating a story, sorry about that.

This is actually useless right now, as we return one `FlightJourney` per one `FlightWatcher`. So right now we return the first result from the list, and we could return multiple, and use this "max count" feature so the users can limit a number those results sent to them.

This needs to be implemented, so I'll create a new story for that. For this PR, withdraw all changes related to this "max count" and just finish the work related to "maxPrice".

LamberM committed:
added new fields in FlightWatcher classes

LamberM committed:
added method in BasePage

added method in BasePage

LamberM committed:
created filtering by cost and count flight

added new field maxPrice

added new field max price

added price filter

prepare for run test

kbannach commented on src/main/java/com/findmyflight/findmyflight/selenium/pages/BasePage.java:
```diff
@@ -23,6 +23,10 @@ protected void waitUntilPresent(By locator) {
         Wait<WebDriver> wait = new WebDriverWait(webDriver, DEFAULT_WAIT_DURATION);
         wait.until(d -> !d.findElements(locator).isEmpty());
     }
+    protected void waitUntilPresent(By locator,Duration duration) {
```
if we add such method, it should be used in `protected void waitUntilPresent(By locator)` with `DEFAULT_WAIT_DURATION` as a parameter, to not duplicate the implementation of the same thing

in general, avoid copy-pasting the code

LamberM replied:
ok, it's done
